### PR TITLE
Modify runner webdriver

### PR DESCRIPTION
**Pull request by abhi2810**
Added Webdriver instantiation logic
Added filter tests

### CODE REVIEW

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java [hunk 1]
```diff
@@ -29,9 +40,44 @@ public String getMethodName() {
...
 
-    public WebDriver getWebDriver(WebDriverConfiguration webDriverConfiguration, Platform platform) {
+    public WebDriver getWebDriver(WebDriverConfiguration webDriverConfiguration, Platform platform) throws MalformedURLException {
         this.webDriverConfiguration = webDriverConfiguration;
-        // instantiate the WebDriver based on whether it is a remote or a local driver
+        switch (webDriverConfiguration.getDriverType()) {
+            case localDriver:
+                System.setProperty(WEBDRIVER_CHROME_DRIVER, Paths.get(platform.getDriverPath()).toString());
+                this.driver = new ChromeDriver();
+                break;
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:49]: - Why only support chrome? Why not also support safari or at least one other driver?
- Move this to it's own function.
- Support passing / setting capabilities for the browser.

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java [hunk 2]
```diff
@@ -29,9 +40,44 @@ public String getMethodName() {
...
+            case remoteDriver:
+                RemoteDriverConfig remoteDriverConfig = webDriverConfiguration.getRemoteDriver();
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:51]: Rename this method to getRemoteDriverConfig

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java [hunk 3]
```diff
@@ -29,9 +40,44 @@ public String getMethodName() {
...
+                    user = System.getenv("BROWSERSTACK_USERNAME");
+                }
+                String accessKey = remoteDriverConfig.getAccessKey();
+                if (StringUtils.isNoneEmpty(System.getenv("BROWSERSTACK_ACCESS_KEY"))) {
+                    accessKey = System.getenv("BROWSERSTACK_ACCESS_KEY");
+                }
+                platformCapabilities.setCapability("browserstack.user", user);
+                platformCapabilities.setCapability("browserstack.key", accessKey);
+                this.driver = new RemoteWebDriver(new URL(remoteDriverConfig.getHubUrl()), platformCapabilities);
+                break;
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:79]: This entire part can be moved to it's own method.

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java [hunk 4]
```diff
@@ -40,13 +86,15 @@ public WebDriver getWebDriver(WebDriverConfiguration webDriverConfiguration, Pla
      */
     protected void succeeded(Description description) {
         LOGGER.info("Succeeded Test :: {} WebDriver Session :: {}", description.getDisplayName(), this.driver);
+        ((JavascriptExecutor) driver).executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \"passed\", \"reason\": \"Test Passed\"}}");
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:89]: Create a CONSTANT variable for this with the values substitutable.

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java [hunk 5]
```diff
@@ -40,13 +86,15 @@ public WebDriver getWebDriver(WebDriverConfiguration webDriverConfiguration, Pla
...
     }
 
     /**
      * Invoked when a test fails
      */
     protected void failed(Throwable e, Description description) {
         LOGGER.info("Failed Test :: {} WebDriver Session :: {}", description.getDisplayName(), this.driver, e);
+        ((JavascriptExecutor) driver).executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \"failed\", \"reason\": \"" + e.getLocalizedMessage() + "\"}}");
```
Comment (anirudha13) [inline src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:97]: Same

**Review by anirudha13 — CHANGES_REQUESTED** [review]

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java (hunk 2 above)
Comment (abhi2810) [reply src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:51]: Done

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java (hunk 1 above)
Comment (abhi2810) [reply src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:49]: done

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java (hunk 5 above)
Comment (abhi2810) [reply src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:97]: done

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java (hunk 4 above)
Comment (abhi2810) [reply src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:89]: done

File: src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java (hunk 3 above)
Comment (abhi2810) [reply src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java:79]: done

**Review by anirudha13 — APPROVED** [review]

**Pull request merged**